pkg/components/state/pluggable.go

```diff
@@ -321,6 +330,46 @@ func (ss *grpcStateStore) Multi(ctx context.Context, request *state.Transactiona
 	return err
 }
 
+// MultiMaxSize returns the maximum number of operations allowed in a transactional request.
+func (ss *grpcStateStore) MultiMaxSize() int {
+	ss.lock.RLock()
+	multiMaxSize := ss.multiMaxSize
+	ss.lock.RUnlock()
+
+	if multiMaxSize != nil {
+		return *multiMaxSize
+	}
+
+	// In a worst case scenario, two or more callers may get to this point,
+	// meaning that we unnecessarily call MultiMaxSize multiple ties,
+	// however this is functionally not a problem and the cost is minimal.
+	ss.lock.Lock()
+	defer ss.lock.Unlock()
```

Comment:
The fix to this "problem" is pretty simple, just check if multiMaxSize is non-nil after acquiring the write lock

```suggestion
	ss.lock.Lock()
	defer ss.lock.Unlock()

	// Check the cached value again in case another goroutine set it
    if multiMaxSize != nil {
		return *multiMaxSize
	}
```